shamir treshold breaks env files (panic: interface conversion: interface {} is float64, not string)

sops: 3.7.1 (ubuntu)

.sops.yaml
```
creation_rules:
- shamir_threshold: 2
  key_groups:
  - pgp:
    - XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  - pgp:
    - YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY
```

command:
`sops example.env`

produces:
```
panic: interface conversion: interface {} is float64, not string

goroutine 1 [running]:
go.mozilla.org/sops/v3/stores/dotenv.(*Store).EmitPlainFile(0x1a17488, 0xc0005d24e0, 0x1, 0x1, 0x3, 0xc0005af040, 0x8, 0x10, 0x0)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/stores/dotenv/store.go:122 +0x3f6
go.mozilla.org/sops/v3/stores/dotenv.(*Store).EmitEncryptedFile(0x1a17488, 0x21816938, 0xed80b80e1, 0x0, 0x11adb59, 0xc, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/stores/dotenv/store.go:107 +0x2fa
main.editTree(0x137d960, 0xc000490fc0, 0x138f380, 0x1a17488, 0x138f380, 0x1a17488, 0xc0001a4840, 0x5c, 0x0, 0xc00048d1d0, ...)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/cmd/sops/edit.go:164 +0xa35
main.editExample(0x137d960, 0xc000490fc0, 0x138f380, 0x1a17488, 0x138f380, 0x1a17488, 0xc0001a4840, 0x5c, 0x0, 0xc00048d1d0, ...)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/cmd/sops/edit.go:84 +0x51b
main.main.func8(0xc0001e4dc0, 0x0, 0x0)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/cmd/sops/main.go:943 +0x24bc
gopkg.in/urfave/cli%2ev1.HandleAction(0xfdafa0, 0x11f8d28, 0xc0001e4dc0, 0x0, 0x0)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:490 +0x82
gopkg.in/urfave/cli%2ev1.(*App).Run(0xc00047c4e0, 0xc00018c000, 0x2, 0x2, 0x0, 0x0)
	/Users/ajvb/gocode/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:264 +0x5f5
main.main()
	/Users/ajvb/gocode/src/go.mozilla.org/sops/cmd/sops/main.go:987 +0x3885

```

--- a/stores/dotenv/store.go
+++ b/stores/dotenv/store.go
@@ -49,7 +49,10 @@ func (store *Store) LoadEncryptedFile(in []byte) (sops.Tree, error) {
 	}
 
 	stores.DecodeNewLines(mdMap)
-	stores.DecodeNonStrings(mdMap)
+	err = stores.DecodeNonStrings(mdMap)
+	if err != nil {
+		return sops.Tree{}, err
+	}
 	metadata, err := stores.UnflattenMetadata(mdMap)
 	if err != nil {
 		return sops.Tree{}, err

--- a/stores/flatten.go
+++ b/stores/flatten.go
@@ -236,13 +236,28 @@ func EncodeNewLines(m map[string]interface{}) {
 }
 
 // DecodeNonStrings will look for known metadata keys that are not strings and decode to the appropriate type
-func DecodeNonStrings(m map[string]interface{}) {
+func DecodeNonStrings(m map[string]interface{}) error {
 	if v, ok := m["mac_only_encrypted"]; ok {
 		m["mac_only_encrypted"] = false
 		if v == "true" {
 			m["mac_only_encrypted"] = true
 		}
 	}
+	if v, ok := m["shamir_threshold"]; ok {
+		switch val := v.(type) {
+			case string:
+				vInt, err := strconv.Atoi(val)
+				if err != nil {
+					return fmt.Errorf("shamir_threshold is not an integer: %s", err.Error())
+				}
+				m["shamir_threshold"] = vInt
+			case int:
+				m["shamir_threshold"] = val
+			default:
+				return fmt.Errorf("shamir_threshold is neither a string nor an integer, but %T", val)
+		}
+	}
+	return nil
 }
 
 // EncodeNonStrings will look for known metadata keys that are not strings and will encode it to strings
@@ -253,6 +268,11 @@ func EncodeNonStrings(m map[string]interface{}) {
 			if vBool {
 				m["mac_only_encrypted"] = "true"
 			}
+		}
+	}
+	if v, found := m["shamir_threshold"]; found {
+		if vInt, ok := v.(int); ok {
+			m["shamir_threshold"] = fmt.Sprintf("%d", vInt)
 		}
 	}
 }

--- a/stores/flatten_test.go
+++ b/stores/flatten_test.go
@@ -222,11 +222,32 @@ func TestDecodeNonStrings(t *testing.T) {
 		{map[string]interface{}{"mac_only_encrypted": "false"}, map[string]interface{}{"mac_only_encrypted": false}},
 		{map[string]interface{}{"mac_only_encrypted": "true"}, map[string]interface{}{"mac_only_encrypted": true}},
 		{map[string]interface{}{"mac_only_encrypted": "something-else"}, map[string]interface{}{"mac_only_encrypted": false}},
+		{map[string]interface{}{"shamir_threshold": "2"}, map[string]interface{}{"shamir_threshold": 2}},
+		{map[string]interface{}{"shamir_threshold": "002"}, map[string]interface{}{"shamir_threshold": 2}},
+		{map[string]interface{}{"shamir_threshold": "123"}, map[string]interface{}{"shamir_threshold": 123}},
+		{map[string]interface{}{"shamir_threshold": 123}, map[string]interface{}{"shamir_threshold": 123}},
 	}
 
 	for _, tt := range tests {
-		DecodeNonStrings(tt.input)
+		err := DecodeNonStrings(tt.input)
+		assert.Nil(t, err)
 		assert.Equal(t, tt.want, tt.input)
+	}
+}
+
+func TestDecodeNonStringsErrors(t *testing.T) {
+	tests := []struct {
+		input map[string]interface{}
+		want  string
+	}{
+		{map[string]interface{}{"shamir_threshold": "foo"}, "shamir_threshold is not an integer: strconv.Atoi: parsing \"foo\": invalid syntax"},
+		{map[string]interface{}{"shamir_threshold": true}, "shamir_threshold is neither a string nor an integer, but bool"},
+	}
+
+	for _, tt := range tests {
+		err := DecodeNonStrings(tt.input)
+		assert.NotNil(t, err)
+		assert.Equal(t, tt.want, err.Error())
 	}
 }
 
@@ -237,6 +258,8 @@ func TestEncodeNonStrings(t *testing.T) {
 	}{
 		{map[string]interface{}{"mac_only_encrypted": false}, map[string]interface{}{"mac_only_encrypted": "false"}},
 		{map[string]interface{}{"mac_only_encrypted": true}, map[string]interface{}{"mac_only_encrypted": "true"}},
+		{map[string]interface{}{"shamir_threshold": 2}, map[string]interface{}{"shamir_threshold": "2"}},
+		{map[string]interface{}{"shamir_threshold": 123}, map[string]interface{}{"shamir_threshold": "123"}},
 	}
 
 	for _, tt := range tests {

--- a/stores/ini/store.go
+++ b/stores/ini/store.go
@@ -188,7 +188,10 @@ func (store *Store) iniSectionToMetadata(sopsSection *ini.Section) (stores.Metad
 		metadataHash[k] = v
 	}
 	stores.DecodeNewLines(metadataHash)
-	stores.DecodeNonStrings(metadataHash)
+	err := stores.DecodeNonStrings(metadataHash)
+	if err != nil {
+		return stores.Metadata{}, err
+	}
 	return stores.UnflattenMetadata(metadataHash)
 }
 

